library/modules/Items.cpp

```diff
@@ -2133,14 +2133,17 @@ int32_t Items::createItem(df::item_type item_type, int16_t item_subtype, int16_t
     delete prod;
 
     DEBUG(items).print("produced %zd items\n", out_items.size());
-    if ( out_items.size() != 1 )
-        return -1;
 
     for (size_t a = 0; a < out_items.size(); a++ ) {
+        // Plant growths need a valid "growth print", otherwise they behave oddly
+        auto growth = virtual_cast<df::item_plant_growthst>(out_items[a]);
+        if (growth)
+            growth->growth_print = growth_print;
```

Comment:
does this fix https://github.com/DFHack/dfhack/issues/898 ?

Reply:
Not by itself - the relevant Lua scripts also need to be updated to pass the correct value here, and I don't know them well enough to do that myself.

Reply:
what is the correct value?

Reply:
can this code be refactored to be included in Items::createItem?
```cpp
        for (size_t i = 0; i < world->raws.plants.all.size(); i++)
        {
            string growths = "";
            df::plant_raw *plant = world->raws.plants.all[i];
            if (plant->id != tokens[0])
                continue;

            for (size_t j = 0; j < plant->growths.size(); j++)
            {
                df::plant_growth *growth = plant->growths[j];
                growths += " " + growth->id;
                if (growth->id != tokens[1])
                    continue;

                // Plant growths specify the actual item type/subtype
                // so that certain growths can drop as SEEDS items
                item_type = growth->item_type;
                item_subtype = growth->item_subtype;
                mat_type = growth->mat_type;
                mat_index = growth->mat_index;

                // Try and find a growth print matching the current time
                // (in practice, only tree leaves use this for autumn color changes)
                for (size_t k = 0; k < growth->prints.size(); k++)
                {
                    df::plant_growth_print *print = growth->prints[k];
                    if (print->timing_start <= *cur_year_tick && *cur_year_tick <= print->timing_end)
                    {
                        growth_print = k;
                        break;
                    }
                }
                // If we didn't find one, then pick the first one (if it exists)
                if (growth_print == -1 && growth->prints.size() > 0)
                    growth_print = 0;
                break;
            }
            if (mat_type == -1)
            {
                if (tokens[1].empty())
                    out.printerr("You must also specify a growth ID.\n");
                else
                    out.printerr("The plant you specified has no such growth!\n");
                out.printerr("Valid growths:%s\n", growths.c_str());
                return CR_FAILURE;
            }
        }
```

Reply:
like, could the discovery of the value of growth_print be moved into Items::createItem so it doesn't have to be passed in?

Reply:
I'd rather not increase the scope of this PR further than necessary.

Reply:
would this be possible in a future PR? do we have enough information with the other parameters (item_type, item_subtype, mat_type, mat_index) to synthesize the growth print?